### PR TITLE
scheduler: refactor hot region scheduler

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Ref #4477. This PR can be reviewed once https://github.com/tikv/pd/pull/4424 is merged.

### What is changed and how it works?

This PR does the following changes:
- remove `xxxPriorityIsBetter`
- move `clearPendingInfluence` to test package
- simplify and rename variables

### Check List

Tests

- Unit test

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by lhy1024** [comment]
Would you like to add more descriptions?

File: server/schedulers/hot_region.go [hunk 1]
```diff
@@ -1103,11 +1062,12 @@ func (bs *balanceSolver) buildOperator() (op *operator.Operator, infl *statistic
 	}
 
 	dim := ""
-	if bs.firstPriorityIsBetter && bs.secondPriorityIsBetter {
+	switch bs.cur.progressiveRank {
```
Comment (bufferflies) [inline server/schedulers/hot_region.go:1065]: Using enum or some dict is better?

File: server/schedulers/hot_region.go [hunk 2]
```diff
@@ -1027,15 +992,9 @@ func stepRank(rk0 float64, step float64) func(float64) int64 {
 }
 
 func (bs *balanceSolver) isReadyToBuild() bool {
```
Comment (bufferflies) [inline server/schedulers/hot_region.go:994]: I think it's difiicult for the judgement condition.

File: server/schedulers/hot_region.go [hunk 3]
```diff
@@ -444,16 +429,8 @@ func (bs *balanceSolver) isValid() bool {
...
-	default:
-		return false
-	}
-	switch bs.opTy {
-	case movePeer, transferLeader:
-	default:
-		return false
-	}
+	// ignore the return value because it will panic if the type is not correct.
+	_ = toResourceType(bs.rwTy, bs.opTy)
```
Comment (bufferflies) [inline server/schedulers/hot_region.go:433]: It returns false if the type is wrong in origin code, and I think `toResourceType` should not  panic.

Comment (rleungx) [reply server/schedulers/hot_region.go:433]: In the original code, it will panic inside the `init` function.

**Review by lhy1024 — APPROVED** [review]

File: server/statistics/store_load.go [hunk 4]
```diff
@@ -158,6 +153,11 @@ func (s *StoreSummaryInfo) AddInfluence(infl *Influence, w float64) {
 	s.PendingSum.Count += infl.Count * w
 }
 
+// IsTiFlash returns if the store is TiFlash.
```
Comment (bufferflies) [inline server/statistics/store_load.go:156]:
```suggestion
// IsTiFlash returns true if the store is TiFlash.
```

**Review by bufferflies — APPROVED** [review]
rest LGTM

**Pull request merged**